### ISSUE TITLE
[Feature Request] Share code on Rust playground

### ISSUE THREAD

**Issue by edwin0cheng** [rust-analyzer/rust-analyzer#8437]
I just saw IntelliJ Rust added this feature recently, and I think it is quite useful and should be not too hard to implement. 

I think the ux flow will be similar to IntelliJ one : 

Make a selection -> `Command Palette` : `Share in Playground` .

**Comment by flodiebold** [comment]
I wonder whether this needs to be in rust-analyzer? It'd need some UI to present the URL, and doesn't need any semantic understanding. E.g. Emacs' rustic-mode already has this, and I'd guess there's a VSCode extension for it as well.

**Comment by Veykril** [comment]
I agree, this seems like a thing a client (or separate extension) can offer.
